lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntimeDynamicTypeResolution.cpp

```diff
@@ -317,11 +317,38 @@ class LLDBTypeInfoProvider : public swift::remote::TypeInfoProvider {
     std::string wrapped;
     // The mangled name passed in is bare. Add global prefix ($s) and type (D).
     llvm::raw_string_ostream(wrapped) << "$s" << mangledName << 'D';
+    swift::Demangle::Demangler dem;
+    auto *node = dem.demangleSymbol(wrapped);
+    if (!node) {
```

Comment:
What does the mangled name look like here to fail demangling? Just a plain "ObjcClass" name?

Reply:
correct, ex `NSObject` or `NSView` etc.